app/components/EventListCompact/index.tsx

```diff
@@ -1,41 +1,39 @@
+import EmptyState from 'app/components/EmptyState';
+import EventItem, { type EventStyle } from 'app/components/EventItem';
+import { Flex } from 'app/components/Layout';
+import LoadingIndicator from 'app/components/LoadingIndicator';
 import type { Event } from 'app/models';
-import EmptyState from '../EmptyState';
-import EventItem, { type EventStyle } from '../EventItem';
-import { Flex } from '../Layout';
-import styles from './EventListCompact.css';
 
 type Props = {
   events: Array<Event>;
   noEventsMessage: string;
-  loggedIn: boolean;
   eventStyle?: EventStyle;
+  loading?: boolean;
 };
 
 const EventListCompact = ({
   events,
   noEventsMessage,
-  loggedIn,
   eventStyle = 'default',
-}: Props) => (
-  <>
-    {events && events.length ? (
-      <Flex column wrap>
-        {events.map((event) => (
-          <EventItem
-            key={event.id}
-            event={event}
-            showTags={false}
-            loggedIn={loggedIn}
-            eventStyle={eventStyle}
-          />
-        ))}
-      </Flex>
-    ) : (
-      <EmptyState>
-        <h2 className={styles.emptyState}>{noEventsMessage}</h2>
-      </EmptyState>
-    )}
-  </>
-);
-
+  loading,
+}: Props) => {
+  return loading ? (
+    <LoadingIndicator margin="20px auto" loading />
+  ) : events && events.length ? (
```

Comment:
I'm more fan of returning this earlier. Turneries can easily become hard to read.

```js
if (loading) {
  return ...
}

return (
  ...
)
```

Reply:
Partially agree, but this is simple enough to be a style preference.

Reply:
Sure, for me it doesn't matter so I'll just change it to returning earlier as you're saying